src/main/scala/com/cognite/sdk/scala/v1/dataModels.scala

```diff
@@ -74,6 +69,24 @@ final case class DataModelInstanceQuery(
     cursor: Option[String] = None
 )
 
+sealed trait PropertyType
+
+sealed trait PropertyTypePrimitive extends PropertyType
+final case class BooleanProperty(value: Boolean) extends PropertyTypePrimitive
+/*object BooleanProperty {
+  def fromJson(json: Json): Option[BooleanProperty] =
+    json.asBoolean.map(BooleanProperty(_))
+}*/
+final case class NumberProperty(value: Double) extends PropertyTypePrimitive
+final case class StringProperty(value: String) extends PropertyTypePrimitive
+
+final case class ArrayProperty[A <: PropertyTypePrimitive](values: Array[A]) extends PropertyType
+
+final case class DataModelInstance(
```

Comment:
I think not only query response but also instance ingestion should use scala types instead of Json. 🤔

Reply:
Also then for ingestion, we should support Null values? And stringified types like `timestamp/datetime/interval` require schema retrieval still :/ Don't know how to fit those into this nice new style

Reply:
yes we will do it for the ingestion as well, so we also need a custom Encoder, not sure about the timestamp/datetime but I think dms just see them as Number/String for now